protocol/java/org/openyolo/protocol/AuthenticationDomain.java

```diff
@@ -74,54 +70,47 @@
     public static AuthenticationDomain getSelfAuthDomain(@NonNull Context context) {
         validate(context, notNullValue(), IllegalArgumentException.class);
 
-        PackageManager pm = context.getPackageManager();
         String packageName = context.getPackageName();
-        PackageInfo packageInfo;
-        try {
-            packageInfo = pm.getPackageInfo(packageName, PackageManager.GET_SIGNATURES);
-        } catch (PackageManager.NameNotFoundException e) {
+        AuthenticationDomain authenticationDomain = fromPackageName(context, packageName);
+        if (null == authenticationDomain) {
             throw new IllegalStateException("Unable to find package info for " + packageName);
         }
 
-        return createAndroidAuthDomain(packageName, packageInfo.signatures[0]);
+        return authenticationDomain;
     }
 
     /**
-     * Creates the list of all authentication domains that represent the specified package.
-     * The list will contain a distinct entry for every signature related to the application.
+     * Returns the {@link AuthenticationDomain} for the application installed on the current device
+     * associated with the given package name otherwise {@code null}.
      */
-    @NonNull
-    public static List<AuthenticationDomain> listForPackage(
+    @Nullable
+    public static AuthenticationDomain fromPackageName(
             @NonNull Context context,
-            @Nullable String packageName) {
+            @NonNull String packageName) {
         validate(context, notNullValue(), IllegalArgumentException.class);
+        validate(packageName, notNullOrEmptyString(), IllegalArgumentException.class);
 
         PackageManager pm = context.getPackageManager();
         PackageInfo packageInfo;
-        if (TextUtils.isEmpty(packageName)) {
-            return Collections.emptyList();
-        }
         try {
             packageInfo = pm.getPackageInfo(packageName, PackageManager.GET_SIGNATURES);
-        } catch (PackageManager.NameNotFoundException e) {
-            return Collections.emptyList();
+        } catch (PackageManager.NameNotFoundException ex) {
+            return null;
         }
 
-        ArrayList<AuthenticationDomain> authDomains =
-                new ArrayList<>(packageInfo.signatures.length);
-        for (Signature signature : packageInfo.signatures) {
-            authDomains.add(createAndroidAuthDomain(packageName, signature));
+        if (1 != packageInfo.signatures.length) {
+            return null;
```

Comment:
Can you log a warning for this? If this ever comes up, it's going to be really difficult to debug otherwise.